### ISSUE TITLE
gasnet configure + make install results in bad path

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#6749]
When building a GASNet-enabled release of Chapel via `configure --prefix=$INSTALL_PREFIX`, `make`, `make install` I find that I get a bad path when trying to invoke the resulting compiler, as follows:

```
$INSTALL_PREFIX/share/chapel/1.16/make/Makefile.base:167: $INSTALL_PREFIX/share/chapel/1.16/third-party/amudprun/Makefile.include
```

It appears that the path where `Makefile.include` is installed is actually under `$INSTALL_PREFIX/lib/` rather than `share/`

``chpl --version``: ``chpl Version 1.16.0 pre-release (913c97e)``

**Comment by bradcray** [comment]
@ben-albrecht, @mppf : FYI

**Comment by bradcray** [comment]
Related: What would it take to test this nightly?

**Comment by bradcray** [comment]
Diving into this a bit to see where I could get, I found that the following mod got me past the most immediate error messages:

```
mbp-bradc (master)$ git diff runtime/etc/Makefile.exe
diff --git a/runtime/etc/Makefile.exe b/runtime/etc/Makefile.exe
index 644f77c..ec8760a 100644
--- a/runtime/etc/Makefile.exe
+++ b/runtime/etc/Makefile.exe
@@ -31,7 +31,7 @@ ifneq ($(SKIP_COMPILE_LINK),skip)
        $(LD) $(GEN_LFLAGS) $(COMP_GEN_LFLAGS) -o $(TMPBINNAME) -L$(CHPL_RT_LIB_DIR) $(TMPBINNAME).o $(CHPLUSEROBJ) $(CHPL_RT_LIB_DIR)/main.o $(CHPL_CL_OBJS) -lchpl -lm $(LIBS) $(CHPL_MAKE_THIRD_PARTY_LINK_ARGS) $(CHPL_MAKE_BASE_LFLAGS)
 endif
 ifneq ($(CHPL_MAKE_LAUNCHER),none)
-       $(MAKE) -f $(CHPL_MAKE_HOME)/runtime/etc/Makefile.launcher all CHPL_MAKE_HOME=$(CHPL_MAKE_HOME) TMPBINNAME=$(TMPBINNAME) BINNAME=$(BINNAME) TMPDIRNAME=$(TMPDIRNAME)
+       $(MAKE) -f $(CHPL_MAKE_HOME)/runtime/etc/Makefile.launcher all CHPL_MAKE_HOME=$(CHPL_MAKE_HOME) TMPBINNAME=$(TMPBINNAME) BINNAME=$(BINNAME) TMPDIRNAME=$(TMPDIRNAME) CHPL_MAKE_THIRD_PARTY=$(CHPL_MAKE_THIRD_PARTY)
 endif
 ifneq ($(TMPBINNAME),$(BINNAME))
        rm -f $(BINNAME)
```

But then I ran into a similar problem with the -I paths used:

```
/Library/Developer/CommandLineTools/usr/bin/make -f /Users/bradc/chapel-install/share/chapel/1.16/runtime/etc/Makefile.launcher all CHPL_MAKE_HOME=/Users/bradc/chapel-install/share/chapel/1.16 TMPBINNAME=output/a.out.tmp BINNAME=a.out TMPDIRNAME=output CHPL_MAKE_THIRD_PARTY=/Users/bradc/chapel-install//lib/chapel/1.16/third-party
echo "*** CHPL_MAKE_HOME=/Users/bradc/chapel-install/share/chapel/1.16 ***"
*** CHPL_MAKE_HOME=/Users/bradc/chapel-install/share/chapel/1.16 ***
rm -f a.out_real
mv output/a.out.tmp a.out_real
echo "#include \"chplcgfns.h\"" > output/launcher_support.c
echo "#include \"config.h\"" >> output/launcher_support.c
echo "#include \"_config.c\"" >> output/launcher_support.c
echo "#include \"chpl_compilation_config.c\"" >> output/launcher_support.c
echo "const char launcher_real_suffix[] = \"_real\";" >> output/launcher_support.c
echo "const char launcher_exe_suffix[] = \"\";" >> output/launcher_support.c
clang  -D_POSIX_C_SOURCE  -fno-strict-overflow  -c -o output/a.out.tmp_launcher.o -I/Users/bradc/chapel-install/share/chapel/1.16/runtime/include/darwin -I/Users/bradc/chapel-install/share/chapel/1.16/runtime/include -I. output/launcher_support.c
output/launcher_support.c:1:10: fatal error: 'chplcgfns.h' file not found
#include "chplcgfns.h"
         ^
1 error generated.
```

Note that both of these problems seem related to Makefile variables not being threaded through to the launcher Makefile.

Also, for context, the thing that ran me into this issue was trying to get gasnet built for the tio site.

**Comment by mppf** [comment]
I'm additionally working on getting a combined quickstart + gasnet build working. Starting with no environment variables set (ie don't setchplenv):
```
export CHPL_COMM=gasnet
make
export CHPL_COMM=none
./configure --prefix=/tmp/p
make
make install
make check
```

We should recommend to tio.run that they do this:
```
export CHPL_TARGET_ARCH=native
export CHPL_COMM=gasnet
make
export CHPL_COMM=none
./configure --prefix=/tmp/p
make
make install
make check
```

I verified that both of the above worked by setting PATH appropriately (and no other variables) and then I ran
```
chpl hello.chpl && ./a.out
chpl --comm=gasnet hello.chpl && ./a.out -nl 2
```

**Comment by bradcray** [comment]
Thanks for jumping on this, Michael.

> I'm additionally working on getting a combined quickstart + gasnet install working. 

What does "quickstart" mean to you here?  Normally I think of it as meaning "what you get when you source the quickstart/setchplenv.bash scripts."  Here, you seem to be suggesting it as a "don't set anything and get the defaults" build (which is what I think tio is doing and what we'd like them to do).  The kind of difference being that the quickstart build will disable a bunch of things where the "don't set anything" build will tend not to (modulo build errors).  Can you clarify?

w.r.t, the command sequences you show: since part of the goal of the `./configure` was to make a Chapel build more like what people are used to, it seems odd to have the configure step take place in the middle of your make steps above.  I assume that this is simply so that CHPL_COMM=none becomes the default mode for users of this installation, but would propose configuring before both makes so it looks at least a little less strange (though not completely).  Any objections to proposing that?

Ultimately, it would be cool if our configure script could take a list of options to build (e.g., --comm=gasnet,none) and would build the full matrix of those options.  I think @thomasvandoren made a helper script that would essentially do this so maybe we could build on top of that?  But now I'm getting into a different feature request (and one that's not at all important now).  A challenge would be "which point in the matrix would be the default for the installation?" where my answer would likely be either the first element in each configuration list or the last.

**Comment by mppf** [comment]
re "quickstart" - that was just a think-o. I was talking about the default (qthreads) config.

> I would propose configuring before both makes so it looks at least a little less strange (though not completely). Any objections to proposing that?

Are you saying you want to suggest they do that? If so, does it work?  Or, were you saying "Let's make that work"? (I think it *should* work....)

> Ultimately, it would be cool if our configure script could take a list of options to build

Yeah that'd be cool but I'm not so sure I want to spend more time on BTR.

**Comment by bradcray** [comment]
I think it should work too -- I was going to suggest they do that instead (so the pattern looked a bit more standard / similar to what they're doing) unless you had reasons that we shouldn't.

**Comment by bradcray** [comment]
Thanks Michael!  (who closed this with PR #6772).
